### PR TITLE
WIP: Add IAC for infrastructure and helm chart deployment

### PR DESCRIPTION
**Pull request by zanieb**
## Summary

Adds `pulumi` powered infrastructure as code for deploying Prefect Server to Azure and GCP.

This allows for creation of
- Managed K8s cluster
- Managed SQL instance
- Prefect Server services on K8s

See the [README](https://github.com/PrefectHQ/server/blob/helm-azure/deploy/README.md) for more details

## Importance

This lowers the barrier to users trying to deploy Prefect Server on various Cloud providers. Whilst not everyone will want to use this particular setup, this at least provides a starting point.

## Checklist

This PR:

- ~[ ] adds new tests (if appropriate)~
- [ ] adds a change file in the `changes/` directory (if appropriate)

### CODE REVIEW

File: deploy/pulumi/providers/azure.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        )
+
+        self.service_principal_pwd = azuread.ServicePrincipalPassword(
+            "prefect-server-app-sp-pwd",
+            service_principal_id=self.service_principal.id,
+            end_date="2099-01-01T00:00:00Z",
+            value=get_password("service-principal-password", self.config),
+        )
+
+        existing_network = pulumi.Config.get("existing-network-name")
```
Comment (mashun4ek) [inline deploy/pulumi/providers/azure.py:56]:
```suggestion
        existing_network = self.config.get("existing-network-name")
```

File: deploy/pulumi/Pulumi.default.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  # Cloud provider specific configuration ------------------------------------------------------------------------------
+
+  # Choose a provider to use, one of azure, gcloud, aws
+  prefect-server:provider: gcp
+
+  # Azure ----------------------------------------------------------
+
+  azure:environment: public
+  azure:location: WestUS
+  prefect-server-azure:k8s-node-type: Standard_B1ms
```
Comment (mashun4ek) [inline deploy/pulumi/Pulumi.default.yaml:12]: I would change to Standard_B2ms (2vcpu, 4 GiB)

**Review by mashun4ek — COMMENTED** [review]
Amazing work! 👏🏻👏🏻

File: deploy/pulumi/providers/azure.py (hunk 1 above)
Comment (madkinsz) [reply deploy/pulumi/providers/azure.py:56]: `self.config` trickily points to `prefect-server-azure` whilst `pulumi.Config` points to `prefect-server`

File: deploy/pulumi/Pulumi.default.yaml (hunk 2 above)
Comment (madkinsz) [reply deploy/pulumi/Pulumi.default.yaml]: 👍

**Pull request closed without merging**